game/src/contrib/systems/CollisionSystem.java

```diff
@@ -19,18 +20,48 @@ public CollisionSystem() {
         super(CollideComponent.class);
     }
 
+    /**
+     * checks every CollideEntity with each other CollideEntity and checks them for a collision.
+     *
+     * <p>Some improvements are done so there is only one collision check between two Entities based
+     * on their IDs. When Entity A does collide with Entity B it also means B collides with A.
+     */
     @Override
     public void execute() {
-        getEntityStream()
-                .flatMap(
-                        a ->
-                                getEntityStream()
-                                        .filter(b -> a.id() < b.id())
-                                        .map(b -> buildData(a, b)))
-                .forEach(this::onEnterLeaveCheck);
+        getEntityStream().flatMap(this::createDataPairs).forEach(this::onEnterLeaveCheck);
+    }
+
+    /**
+     * Creates a Stream where every other Entity has a higher id.
+     *
+     * @param a Entity which is the lower id partner
+     * @return the stream which contains every valid pair of Entities
+     */
+    private Stream<CollisionData> createDataPairs(Entity a) {
+        return getEntityStream().filter(b -> isIDSmallerThen(a, b)).map(b -> newDataPair(a, b));
+    }
+
+    /**
+     * Makes sure only Entities with a higher id are paired up.
```

Comment:
das ist aber nicht das, was in dieser funktion passiert - als erster satz passt das also nicht.

aber eine schöne beschreibung, wozu diese funktion benutzt wird, also ein schöner text für den absatz nach der summary.

Reply:
habe es mal versucht umzuschreiben

Reply:
hab mal im neuen kommentar einen vorschlag gemacht :)